meshroom/submitters/rippleSubmitter.py

```diff
@@ -0,0 +1,139 @@
+import os 
+import json
+
+#meshroom modules
+from meshroom.core.desc import Level
+from meshroom.core.submitter import BaseSubmitter
+
+#mpc logging import
+import mpc.logging
+
+#Ripple imports
+from mpc.ripple.rippleConfig import RippleConfig as _RippleConfig
+from mpc.ripple.rippleProcess import RippleProcess
+from mpc.ripple.dispatcher import DefaultDispatcher
+from mpc.ripple.rippleStorage import RippleStorage
+from mpc.ripple.rippleUtilities import RippleGroup
+from mpc.ripple.rippleAttribute import RippleAttribute
+
+#validators for numbers
+from mpc.pyCore.validators import IntValidator
+
+_log = mpc.logging.getLogger()
+
+currentDir = os.path.dirname(os.path.realpath(__file__))
+binDir = os.path.dirname(os.path.dirname(os.path.dirname(currentDir)))
+
+# Give access to min/maxProcessors, which is an alias to slots
+class RippleProcessWithSlots(RippleProcess):
+    minProcessors = RippleAttribute('', IntValidator(), 1, True)
+    maxProcessors = RippleAttribute('', IntValidator(), 1, True)
+
+class RippleSubmitter(BaseSubmitter):
+    def __init__(self, parent=None):
+        super(RippleSubmitter, self).__init__(name='Ripple', parent=parent)
+
+    def createTask(self, meshroomFile, node, parents):
+
+        nbBlocks = 1   
+        
+        #Map meshroom GPU modes to MPC services
+        gpudict = {
+            "NONE":"",
+            "NORMAL":",gpu8G",
+            "INTENSIVE":",gpu16G"
+        }
+        
+        #Specify some constraints
+        requirements = "!\"rs*\",@.mem>25{gpu}".format(gpu=gpudict[node.nodeDesc.gpu.name])
+
+        #specify which node to wait before launching the current one
+        waitsFor = []
+        for parent in parents:
+            waitsFor.append(parent.name)
+
+        #Basic command line for this node
+        command='meshroom_compute --node {nodeName} "{meshroomFile}" --extern'.format(nodeName=node.name, meshroomFile=meshroomFile)
+
+        if node.isParallelized:
+            _, _, nbBlocks = node.nodeDesc.parallelization.getSizes(node)
+
```

Comment:
maxProcessors = (Level.INTENSIVE in (node.nodeDesc.ram, node.nodeDesc.cpu)) ? 4 : 1

Reply:
Done